google/cloud/aiplatform/experimental/vertex_model/base.py

```diff
@@ -147,23 +155,54 @@ def f(*args, **kwargs):
                     display_name="my_training_job",
                     script_path=str(script_path),
                     # programatically determine the dependency in the future
-                    requirements=["pandas>=1.3"],
+                    requirements=["pandas>=1.3", "torch>=1.7"],
                     # https://cloud.google.com/vertex-ai/docs/training/pre-built-containers
                     container_uri="us-docker.pkg.dev/vertex-ai/training/pytorch-xla.1-7:latest",
                 )
 
                 # In the custom training job, a MODEL directory will be provided as an env var
                 # our code should serialize our MODEL to that directory
 
-                obj._training_job.run(replica_count=1)
+                obj._model = obj._training_job.run(
+                    model_display_name="my_model",
+                    replica_count=1,
+                    machine_type="n1-standard-4",
+                    accelerator_type="NVIDIA_TESLA_K80",
+                    accelerator_count=1,
+                )
 
     return f
 
 
+def vertex_predict_function_wrapper(method: Callable[..., Any]):
+    @functools.wraps(method)
+    def p(*args, **kwargs):
+        obj = method.__self__
+
+        # Local predictions can be made
+        if method.__self__.training_mode == "local" or obj._model is None:
+            return method(*args, **kwargs)
+
+        # Assuming only local prediction for now
+        output_dir = obj._model._gca_resource.artifact_uri
+        model_uri = pathlib.Path(output_dir) / (
+            "my_" + obj.training_mode + "_model.pth"
+        )
+
+        my_model = model._deserialize_remote_model(str(model_uri))
```

Comment:
To avoid deserializing at every predict call, you can assign my_model as an instance attribute.